### ISSUE TITLE
GUI appears tiny on high resolution screen

### ISSUE THREAD

**Issue by MFairley** [timholy/ProfileView.jl#153]
The GUI appears strange to me with the chart displayed only on the bottom left of the window. I've attached a screenshot generated using the example code in the Readme. I guess this could be a problem with high resolution displays. I am running mac OS 10.15.6 with a 27-inch (5120 x 2880) screen.

<img width="818" alt="Screen Shot 2020-09-17 at 3 46 13 PM" src="https://user-images.githubusercontent.com/4374785/93527179-17a08980-f8fe-11ea-9682-c0260baa179a.png">

**Comment by timholy** [comment]
It *might* be a GUI issue, but it also might be a `libunwind` issue (which we use to gather stacktraces). Here's a demo:
```julia
julia> using Profile, LinearAlgebra

julia> A = rand(300, 300);

julia> svd(A);

julia> @profile svd(A);

julia> Profile.print()
Overhead ╎ [+additional indent] Count File:Line; Function
=========================================================
  ╎47 @Base/client.jl:485; _start()
  ╎ 47 @Base/client.jl:302; exec_options(opts::Base.JLOptions)
  ╎  47 @Base/client.jl:372; run_main_repl(interactive::Bool,...
  ╎   47 @Base/essentials.jl:706; invokelatest
  ╎    47 @Base/essentials.jl:707; #invokelatest#2
  ╎     47 @Base/client.jl:387; (::Base.var"#848#850"{Bool, Bo...
  ╎    ╎ 47 @REPL/src/REPL.jl:304; run_repl(repl::REPL.AbstractRE...
  ╎    ╎  47 @REPL/src/REPL.jl:316; run_repl(repl::REPL.AbstractR...
  ╎    ╎   47 @REPL/src/REPL.jl:185; start_repl_backend(backend::...
  ╎    ╎    47 @REPL/src/REPL.jl:200; repl_backend_loop(backend::R...
  ╎    ╎     47 @REPL/src/REPL.jl:139; eval_user_input(ast::Any, b...
 3╎    ╎    ╎ 47 @Base/boot.jl:345; eval(m::Module, e::Any)
  ╎    ╎    ╎  44 ...gebra/src/svd.jl:157; svd(A::Matrix{Float64})
  ╎    ╎    ╎   44 ...gebra/src/svd.jl:157; #svd#85
  ╎    ╎    ╎    44 ...gebra/src/svd.jl:93; svd!##kw
  ╎    ╎    ╎     44 ...ebra/src/svd.jl:97; svd!(A::Matrix{Float64}; ...
  ╎    ╎    ╎    ╎ 44 ...ebra/src/svd.jl:104; _svd!
44╎    ╎    ╎    ╎  44 .../src/lapack.jl:1642; gesdd!(job::Char, A::M...
Total snapshots: 60
```
Wait?!? 60 snapshots? I only see 47!

Now this:
```julia
julia> Profile.print(C=true; maxdepth=20)
Overhead ╎ [+additional indent] Count File:Line; Function
=========================================================
 ╎47 /home/tim/src/julia-master/julia:?; _start
 ╎ 47 /lib/x86_64-linux-gnu/libc.so.6:?; __libc_start_main
 ╎  47 /home/tim/src/julia-master/ui/repl.c:227; main
 ╎   47 /home/tim/src/julia-master/ui/repl.c:106; true_main
 ╎    47 /home/tim/src/julia-master/ui/../src/julia.h:1682; jl_apply
 ╎     47 /home/tim/src/julia-master/src/gf.c:2345; jl_apply_generic
 ╎    ╎ 47 /home/tim/src/julia-master/src/gf.c:2163; _jl_invoke
 ╎    ╎  47 /home/tim/src/julia-master/usr/lib/julia/sys.so:?; jfptr__start_34615
 ╎    ╎   47 @Base/client.jl:485; _start()
 ╎    ╎    47 @Base/client.jl:302; exec_options(opts::Base.JLOptions)
 ╎    ╎     47 @Base/client.jl:372; run_main_repl(interactive::Bool, quiet::Bool, banner::Bool, history_file::Bool, color_set::Bool)
 ╎    ╎    ╎ 47 @Base/essentials.jl:706; invokelatest
 ╎    ╎    ╎  47 @Base/essentials.jl:707; #invokelatest#2
 ╎    ╎    ╎   47 /home/tim/src/julia-master/src/builtins.c:722; jl_f__apply_latest
 ╎    ╎    ╎    47 /home/tim/src/julia-master/src/builtins.c:672; do_apply
 ╎    ╎    ╎     47 /home/tim/src/julia-master/src/julia.h:1682; jl_apply
 ╎    ╎    ╎    ╎ 47 /home/tim/src/julia-master/src/gf.c:2345; jl_apply_generic
 ╎    ╎    ╎    ╎  47 /home/tim/src/julia-master/src/gf.c:2163; _jl_invoke
 ╎    ╎    ╎    ╎   47 /home/tim/src/julia-master/usr/lib/julia/sys.so:?; jfptr_YY.848_32195
 ╎    ╎    ╎    ╎    47 @Base/client.jl:387; (::Base.var"#848#850"{Bool, Bool, Bool})(REPL::Module)
 ╎    ╎    ╎    ╎     47 /home/tim/src/julia-master/src/gf.c:2345; jl_apply_generic
1╎1  /home/tim/src/julia-master/usr/bin/../lib/libLLVM-9jl.so:?; _ZN4llvm16DenseMapIteratorIPKvPKNS_8PassInfoENS_12DenseMapInfoIS2_EENS_6detail12DenseMapPairIS2_S5_EELb1EEC2E...
1╎1  /home/tim/src/julia-master/usr/bin/../lib/libLLVM-9jl.so:?; _ZNK4llvm12AttributeSet12getAttributeENS_9StringRefE
2╎2  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; dcopy_k_HASWELL
3╎3  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; dgemm_kernel_HASWELL
1╎1  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; dnrm2_k_HASWELL
 ╎5  [unknown stackframe]
 ╎ 1  /home/tim/src/julia-master/usr/bin/../lib/libgfortran.so.4:?; _gfortran_concat_string
1╎ 1  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; dswap_k_HASWELL
1╎ 1  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; exec_blas
 ╎ 2  [unknown stackframe]
 ╎  2  /lib/x86_64-linux-gnu/libc.so.6:?; posix_memalign
2╎   2  /home/tim/src/julia-master/usr/bin/../lib/libopenblas64_.so:?; disnan_64_
Total snapshots: 60
```
The oddity is that the ones at the end, which come from various C & Fortran libraries we call, are not attributed to any Julia statements---they are effectively detached. It's quite possible that the big empty white space consists of a huge number of detached samples; try viewing with `C=true` and see if that explains it.

If that explains your finding, then it's a libunwind/Julia issue, not a ProfileView issue. I guess we could trim them but that's misleading in its own way; that code *did* take time, the only issue is we've lost the ability to attribute it to its callers.

**Comment by MFairley** [comment]
Looks like your hypothesis is true. Here's a screenshot with `ProfileView.view(C=true)`
<img width="823" alt="Screen Shot 2020-09-17 at 5 29 15 PM" src="https://user-images.githubusercontent.com/4374785/93534805-61dc3780-f90b-11ea-9a3b-7ae8dcceb0c0.png">

**Comment by MFairley** [comment]
Also, I just tried it again with `julia -t 1`, Previously I was using 8 threads. Now there is much less blank space but still some.

**Comment by ianshmean** [comment]
I see the same for
```
@profile Pkg.precompile()
ProfileView.view(C=true)
```
![image](https://user-images.githubusercontent.com/1694067/94745111-20567e00-0348-11eb-8b68-38b1d5710ec0.png)

**Comment by ianshmean** [comment]
`C=true` seems necessary even for more basic tests

```
@profile [sleep(0.1) for i in 1:10]
ProfileView.view(C=true)
```
![image](https://user-images.githubusercontent.com/1694067/94747852-f1430b00-034d-11eb-8e7f-9bcd55325e6a.png)

**Comment by timholy** [comment]
@IanButterworth, is it fair to say this has been fixed on julia nightly?

**Comment by IanButterworth** [comment]
It was actually fixed in MacOS Monterey 12.0.0 itself, so on that MacOS release all julia versions are fixed
